### PR TITLE
Remove potential race condition in onfini handler registration

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/coro.py [hunk 1]
```diff
@@ -148,6 +148,19 @@ async def ornot(func, *args, **kwargs):
         return await retn
     return retn
 
+bgtasks = set()
```
Comment (MichaelSquires) [inline synapse/lib/coro.py:151]: Is this just to keep a reference ot the task so it runs?

Comment (invisig0th) [reply synapse/lib/coro.py:151]: Yep. Otherwise it can be ripped out of the ioloop.

File: synapse/telepath.py
Comment (MichaelSquires) [inline synapse/telepath.py:1]: Did you mean to include this file in the PR? Seems unrelated to the onfini handler.

**Review by MichaelSquires — APPROVED** [review]
